hail/python/hail/utils/backend.py

```diff
@@ -0,0 +1,27 @@
+import abc
+
+from hail.utils.java import *
+from hail.expr.types import HailType
+from hail.ir.renderer import Renderer
+
+
+class Backend(object):
+    @abc.abstractmethod
+    def interpret(self, ir):
+        return
+
+
+class SparkBackend(Backend):
+    def interpret(self, ir):
+        assert isinstance(ir, hail.ir.IR)
+
+        r = Renderer(stop_at_jir=False)
```

Comment:
Same here, use stop_at_jir=True.

Reply:
Won't that truncate the IR though prematurely and the full thing won't be interpreted?

Reply:
No.  That just means you use a JavaTable to refer to a IR pointer rather than rendering the IR in Python and-reparsing it in  Scala.  The jir pointed to by an IR should be the same as the result of rendering and parsing that IR.  So the behavior is exactly the same, but you save quadratic parsing.